base/array.jl

```diff
@@ -15,7 +15,8 @@ typealias StridedVector{T,A<:DenseArray,I<:Tuple{Vararg{RangeIndex}}}  Union(Den
 typealias StridedMatrix{T,A<:DenseArray,I<:Tuple{Vararg{RangeIndex}}}  Union(DenseArray{T,2}, SubArray{T,2,A,I})
 typealias StridedVecOrMat{T} Union(StridedVector{T}, StridedMatrix{T})
 
-call{T}(::Type{Vector{T}}, m::Integer) = Array{T}(m)
+call{T}(::Type{Vector{T}}, m::Integer=0) = Array{T}(m)
```

Comment:
I guess your way would be preferable if we were to want to generalize to `Matrix` as well, because then we would not want a single-argument constructor to work.  Is there any other reason to prefer the method you suggest?